### PR TITLE
Feature/refactor to hooks i

### PR DESCRIPTION
**Pull request by Matt-Roden**
# Description drawing

- Refactored all existing components from class based to functional that use hooks
- Currently have error and validate pokemon method
- --> Error: On an invalid search, the conditional to show the error message to the user is not display, instead the app is breaking and sending down undefined as a prop to the PokemonDetails component 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] local host 3000
- [x] console.log()
- [x] dev tools
- [ ] cypress testing

# Checklist:

- [x] My code follows the style guidelines of this project/linted
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -1,42 +1,41 @@
-import React, { Component } from "react";
+import React, { useEffect, useState } from 'react'
```
Comment (hayleyw7) [inline src/components/App/App.js:1]: Hooks!

**Review by hayleyw7 — APPROVED** [review]
Thanks for getting these set up. I'm excited to learn more.

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
+      const pokeDexData = await res.json()
+      setPokeDex(pokeDexData.results)
+    } catch (err) {
+      console.log('Error: ', err)
+    }
+  };
+
+  const addPokemon = (queriedPokemon) => {
+    const foundPokemon = validatePokemonQuery(queriedPokemon)
+      setFoundPokemon([foundPokemon])
```
Comment (joanafbrito) [inline src/components/App/App.js:31]: Thank's for that... I like how it is ending up!

File: src/components/App/App.js [hunk 3]
```diff
@@ -50,68 +49,60 @@ class App extends Component {
...
         }
       }
-    });
-
+    })
+    console.log(verifiedName, '<><><><>')
+// This condition not working
     if (verifiedName === undefined) {
-      return this.setState({ error: "Not  a valid Name or id , try again" });
+      setError("Not  a valid Name or id , try again");
```
Comment (joanafbrito) [inline src/components/App/App.js:56]: We definitely  need to figure the why here... we will do it! Let's rock this one as well

**Review by joanafbrito — APPROVED** [review]
Thanks for doing that, let's go to .gsap()

**Pull request merged**